test/service/test_users.py

```diff
@@ -0,0 +1,205 @@
+            users.authenticate_via_api_key('test-api_key')
+        
+    def test_throws_on_failed_auth(self):
+        self.mock_get_user_by_api_key.return_value.fetchone.return_value = None
+        with self.assertRaises(users.BeachfrontUnauthorized):
+            new_user = users.authenticate_via_api_key('test-api_key')
+        
+    def test_verifies_api_key(self):
+        self.mock_get_user_by_api_key.return_value.fetchone.return_value = create_user_db_record()
+        with self.assertRaises(users.DatabaseMismatchError):
+            new_user = users.authenticate_via_api_key('not-the-returned-api_key')
+        
+    def test_return_good_user_id(self):
+        self.mock_get_user_by_api_key.return_value.fetchone.return_value = create_user_db_record(api_key='test-api_key')
+        new_user = users.authenticate_via_api_key('test-api_key')
+        self.assertIsInstance(new_user, users.User)
+        self.assertEqual('test-user-id', new_user.geoaxis_uid)
+
+    def test_return_good_user_name(self):
+        self.mock_get_user_by_api_key.return_value.fetchone.return_value = create_user_db_record(api_key='test-api_key')
+        new_user = users.authenticate_via_api_key('test-api_key')
+        self.assertIsInstance(new_user, users.User)
+        self.assertEqual('test-user-name', new_user.user_name)
+
+    def test_return_good_api_key(self):
+        self.mock_get_user_by_api_key.return_value.fetchone.return_value = create_user_db_record(api_key='test-api_key')
+        new_user = users.authenticate_via_api_key('test-api_key')
+        self.assertIsInstance(new_user, users.User)
+        self.assertEqual('test-api_key', new_user.bf_api_key)
+
+class GetByIdTest(unittest.TestCase):
+    def setUp(self):
+        self._mockdb = helpers.mock_database()
+        self._logger = logging.getLogger('bfapi.service.users')
+        self._logger.disabled = True
+        self.mock_get_user = self.create_mock('bfapi.db.users.select_user')
+
+    def tearDown(self):
+        self._mockdb.destroy()
+        self._logger.disabled = False
+
+    def create_mock(self, target_name):
+        patcher = patch(target_name)
+        self.addCleanup(patcher.stop)
+        return patcher.start()
+        
+    def test_throws_on_db_error(self):
+        self.mock_get_user.side_effect = helpers.create_database_error()
+        with self.assertRaises(DatabaseError):
+            users.get_by_id('test-user-id')
+        
+    def test_return_none_if_none(self):
+        self.mock_get_user.return_value.fetchone.return_value = None
+        new_user = users.get_by_id('test-user-id')
+        self.assertEqual(None, new_user)
+        
+    def test_throws_on_wrong_id(self):
+        self.mock_get_user.return_value.fetchone.return_value = create_user_db_record()
+        with self.assertRaises(users.DatabaseMismatchError):
+            users.get_by_id('not-the-returned-geoaxis-id')
+
+    def test_return_good_user_id(self):
```

Comment:
Can you rename this to `test_assigns_correct_*`?

Ref:
https://github.com/venicegeo/bf-api/blob/master/test/service/test_productlines.py#L412